featuretools/primitives/standard/datetime_transform_primitives.py

```diff
@@ -488,3 +490,47 @@ def get_function(self):
         def year(vals):
             return vals.dt.year
         return year
+
+
+class IsFederalHoliday(TransformPrimitive):
+    """Determines if a given datetime is a federal holiday.
+    Description:
+        This primtive currently only works for the United States
+        and Canada with dates between 1950 and 2100.
+    Args:
+        country (str): Country to use for determining Holidays.
+            Default is 'US'. Should be one of the available countries here:
+            https://github.com/dr-prodigy/python-holidays#available-countries
+    Examples:
+        >>> from datetime import datetime
+        >>> is_federal_holiday = IsFederalHoliday(country="US")
+        >>> is_federal_holiday([
+        ...     datetime(2019, 7, 4, 10, 0, 30),
+        ...     datetime(2019, 2, 26)]).tolist()
+        [True, False]
+    """
+    name = "is_federal_holiday"
+    input_types = [ColumnSchema(logical_type=Datetime)]
+    return_type = ColumnSchema(logical_type=BooleanNullable)
+
+    def __init__(self, country='US'):
+        self.country = country
+        try:
+            self.holidays = holidays.CountryHoliday(self.country)
```

Comment:
in version 0.13, `CountryHoliday` was deprecated in favor of `country_holidays`

do we want to bump our min requirement to 0.13, use the new API, and drop the `KeyError` exception in favor of `NotImplementedError`?

Reply:
sure we can do that

Reply:
If we do that now, should we also fix the other two primitives that use `holidays` and are catching both error types?